examples/tests/backend_driver.py

```diff
@@ -30,313 +30,23 @@
-    'text_rotation_relative_to_line.py',
-    'transoffset.py',
-    'xcorr_demo.py',
-    'zorder_demo.py',
-    ]
-
-
-files['api'] = [
-    'agg_oo.py',
-    'barchart_demo.py',
-    'bbox_intersect.py',
-    'collections_demo.py',
-    'colorbar_only.py',
-    'custom_projection_example.py',
-    'custom_scale_example.py',
-    'date_demo.py',
-    'date_index_formatter.py',
-    'donut_demo.py',
-    'font_family_rc.py',
-    'image_zcoord.py',
-    'joinstyle.py',
-    'legend_demo.py',
-    'line_with_text.py',
-    'logo2.py',
-    'mathtext_asarray.py',
-    'patch_collection.py',
-    'quad_bezier.py',
-    'scatter_piecharts.py',
-    'span_regions.py',
-    'two_scales.py',
-    'unicode_minus.py',
-    'watermark_image.py',
-    'watermark_text.py',
-]
-
-files['units'] = [
-    'annotate_with_units.py',
-    #'artist_tests.py',  # broken, fixme
-    'bar_demo2.py',
-    #'bar_unit_demo.py', # broken, fixme
-    #'ellipse_with_units.py',  # broken, fixme
-    'radian_demo.py',
-    'units_sample.py',
-    #'units_scatter.py', # broken, fixme
-
-    ]
-
-files['mplot3d'] = [
-    '2dcollections3d_demo.py',
-    'bars3d_demo.py',
-    'contour3d_demo.py',
-    'contour3d_demo2.py',
-    'contourf3d_demo.py',
-    'lines3d_demo.py',
-    'polys3d_demo.py',
-    'scatter3d_demo.py',
-    'surface3d_demo.py',
-    'surface3d_demo2.py',
-    'text3d_demo.py',
-    'wire3d_demo.py',
-    ]
-
-# dict from dir to files we know we don't want to test (e.g., examples
-# not using pyplot, examples requiring user input, animation examples,
-# examples that may only work in certain environs (usetex examples?),
-# examples that generate multiple figures
-
-excluded = {
-    'units': ['__init__.py', 'date_support.py', ],
-}
-
-
-def report_missing(dir, flist):
-    'report the py files in dir that are not in flist'
-    globstr = os.path.join(dir, '*.py')
-    fnames = glob.glob(globstr)
-
-    pyfiles = {os.path.split(fullpath)[-1] for fullpath in set(fnames)}
-
-    exclude = set(excluded.get(dir, []))
-    flist = set(flist)
-    missing = list(pyfiles - flist - exclude)
-    if missing:
-        print('%s files not tested: %s' % (dir, ', '.join(sorted(missing))))
-
-
-def report_all_missing(directories):
-    for f in directories:
-        report_missing(dirs[f], files[f])
-
-
-# tests known to fail on a given backend
-
-failbackend = dict(
-    svg=('tex_demo.py', ),
-    agg=('hyperlinks.py', ),
-    pdf=('hyperlinks.py', ),
-    ps=('hyperlinks.py', ),
-    )
-
+for subdir, _, f in os.walk('../'):
+    if subdir not in ['../']:
+        dirs[subdir[3:]] = subdir
+        files[subdir[3:]] = [s for s in f if s[-3:] == '.py']
+
+# list of whole directories to not run
+excluded_dirs = ['user_interfaces']
+# dict of files to not run on any backend
+excluded_files = {'units': ['date_support.py'],
+                  'widgets': ['lasso_selector_demo.py']}
+# dict of files to not run on specific backend
+failbackend = {'svg': ['tex_demo.py'],
```

Comment:
I'd break after `{` and put every line only one indent in instead of aligned to the `{`, but that's a matter of taste.